projects/igniteui-angular/src/lib/tabbar/tabbar.component.ts

```diff
@@ -402,47 +514,118 @@ export class IgxTabComponent {
     get changesCount(): number {
         return this._changesCount;
     }
+
+    private _disabled = false;
+
     /**
      * Gets whether the tab is disabled.
      * ```typescript
      * let isDisabled = this.tab.disabled;
      * ```
      * @memberof IgxTabComponent
      */
+    @Input()
     get disabled(): boolean {
-        const panel = this.relatedPanel;
-
-        if (panel) {
-            return panel.disabled;
+        return this.relatedPanel ? this.relatedPanel.disabled : this._disabled;
+    }
+    set disabled(newValue: boolean) {
+        if (this.relatedPanel) {
+            this.relatedPanel.disabled = newValue;
+        } else {
+            this._disabled = newValue;
         }
     }
+
+    public _selected = false;
+
     /**
      * Gets whether the tab is selected.
      * ```typescript
      * let isSelected  = this.tab.isSelected;
      * ```
      * @memberof IgxTabComponent
      */
+    @Input()
+    set isSelected(newValue: boolean) {
+        if (this.relatedPanel) {
+            this.relatedPanel.isSelected = newValue;
+        } else if (this._selected !== newValue) {
+            this._selected = newValue;
+            if (this._selected) {
+                this._tabBar.onTabSelected.emit({ tab: this, panel: null });
+            }
+        }
+    }
     get isSelected(): boolean {
-        const panel = this.relatedPanel;
+        return this.relatedPanel ? this.relatedPanel.isSelected : this._selected;
+    }
 
-        if (panel) {
-            return panel.isSelected;
-        }
+    @HostBinding('class.igx-bottom-nav__menu-item--selected')
+    public get cssClassSelected(): boolean {
+        return this.isSelected;
+    }
+
+    @HostBinding('class.igx-bottom-nav__menu-item--disabled')
+    public get cssClassDisabled(): boolean {
+        return this.disabled;
+    }
+
+    @HostBinding('class.igx-bottom-nav__menu-item')
+    public get cssClass(): boolean {
+        return (!this.disabled && !this.isSelected);
     }
+
     /**
      * Gets the `index` of the tab.
      * ```typescript
      * let tabIndex = this.tab.index;
      * ```
      * @memberof IgxTabComponent
      */
-    get index(): number {
-        return this._tabBar.tabs.toArray().indexOf(this);
+    public get index(): number {
+        if (this._tabBar && this._tabBar.hasContentTabs) {
+            return this._tabBar.contentTabs.toArray().indexOf(this);
+        }
+        if (this._tabBar && this._tabBar.tabs) {
+            return this._tabBar.tabs.toArray().indexOf(this);
+        }
+        return 0;
+    }
+
+    /**@hidden*/
+    @ViewChild('defaultTabTemplate', { read: TemplateRef })
+    protected defaultTabTemplate: TemplateRef<any>;
+
+    /**
+     * Returns the `template` for this IgxTabComponent.
+     * ```typescript
+     * let tabItemTemplate = this.tabItem.template;
+     * ```
+     * @memberof IgxTabComponent
+     */
+    public get template(): TemplateRef<any> {
```

Comment:
Consider adding template setter in order to customize the tab template when the tab is content child.

Reply:
Addressed.